vignettes/query-RESULTS-silviculture-data-using-bcdata.Rmd

```diff
@@ -0,0 +1,213 @@
+  filter(INTERSECTS(dpg)) %>% #filter for records that are within the DPG boundary
+  filter(
+      S_SPECIES_CODE_1 %in% spp_list |  #filter for LW records 
+      S_SPECIES_CODE_2 %in% spp_list |
+      S_SPECIES_CODE_3 %in% spp_list |
+      S_SPECIES_CODE_4 %in% spp_list |
+      S_SPECIES_CODE_5 %in% spp_list
+  ) %>% 
+  collect() #collect/download the data
+```
+
+
+### Exploring the Data
+
+Let's look at the dimensions of this now much more manageable data object we have downloaded from the B.C. Data Catalogue:
+
+```{r dim-data}
+dim(trees_dpg)
+```
+
+We can see there are `r dim(trees_dpg)[1]` treatment units planted with western larch, and we can make a quick map of these harvested openings for the Prince George Natural Resource District:
+
+```{r map-larch-plantations-dpg, message = FALSE}
+trees_dpg %>% 
+  ggplot() + 
+  geom_sf() + 
+  geom_sf(data = dpg, fill = NA) + #add the DPG spatial boundary
+  theme_minimal()
+```
+
+
+We can also create some quick descriptive summaries of the data, treating the geospatial attribute table as a data frame in R, and answer our original question---how much western larch has been planted in the Prince George Natural Resource District?
+
+
+#### What is the size and age distribution of larch plantations in the Prince George Natural Resource District in the year 2020?
+
+```{r, warning = FALSE}
+trees_dpg %>% 
+  mutate(age = 2020 - REFERENCE_YEAR + S_SPECIES_AGE_1) %>% #create a plantation age column
+  ggplot() +  #start a plot
+  aes(x = age, y = FEATURE_AREA_SQM/10000) + #convert feature area to hectares
+  geom_bar(stat = "sum") +
+  scale_x_continuous(name = "Stand Age (years)",
+                     limits = c(2, 30),
+                     breaks = seq(from = 5, to = 30, by = 5)) +
+  labs(y = "Sum Treatment Unit Area (ha)",
+       title = "Area Planted with Western Larch by Stand Age\nin Prince George Natural Resource District",
+       caption = paste0("Data sourced from the B.C. Data Catalogue\n on ",
+                        Sys.Date(),
+                        " using the bcdata R package")) +
+  theme_minimal() +
+  theme(legend.position = "none")
+```
+
+#### What is the Biogeoclimatic Ecosystem Classification (BEC) distribution of western larch plantations in the Prince George Natural Resource District?
+
+We can download [British Columbia biogeoclimatic (BEC) data](https://catalogue.data.gov.bc.ca/dataset/bec-map) from the B.C. Data Catalogue using `bcdata` and join to our existing `trees_dpg` geospatial dataset using the `st_join` function from the `sf` package. You can learn more about `sf` [here](https://r-spatial.github.io/sf/).
+
+
+```{r, cache = TRUE}
+library(sf) #load the sf package
+
+# Load the BEC data for Prince George Natural Resource District
+bgc_dpg <- bcdc_query_geodata("WHSE_FOREST_VEGETATION.BEC_BIOGEOCLIMATIC_POLY") %>%
+  filter(BBOX(st_bbox(dpg), crs = `ESP:3005`)) %>% #note filtering with a BoundingBox
```

Comment:
After #200 we can now simplify this to:
```suggestion
  filter(BBOX(dpg, crs = `EPSG:3005`)) %>% #note filtering with a BoundingBox
```

Reply:
But only with the dev version of `bcdata`?

Reply:
Yeah only the dev version.

Reply:
So we time the release of the vignette with a CRAN release that includes #200?

Reply:
Yeah I think this is all going into the next release, right?